tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/model/ParentSDKProjectImportOpMethods.java

```
@@ -0,0 +1,150 @@
+import com.liferay.ide.sdk.core.SDKUtil;
+
+import java.util.Map;
+
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.MultiStatus;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.sapphire.modeling.Path;
+import org.eclipse.sapphire.modeling.ProgressMonitor;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.PathBridge;
+import org.eclipse.sapphire.platform.ProgressMonitorBridge;
+import org.eclipse.sapphire.platform.StatusBridge;
+
+/**
+ * @author Simon Jiang
+ */
+public class ParentSDKProjectImportOpMethods
+{
+
+    public static final Status execute( final ParentSDKProjectImportOp op, final ProgressMonitor pm )
+    {
+        final IProgressMonitor monitor = ProgressMonitorBridge.create( pm );
+
+        monitor.beginTask( "Importing Liferay plugin projects...", 100 );
+
+        Status retval = Status.createOkStatus();
+
+        final Path sdkLocation = op.getSdkLocation().content();
+
+        if( sdkLocation == null || sdkLocation.isEmpty() )
+        {
+            return Status.createErrorStatus( "SDK folder cannot be empty" );
+        }
+
+        final Job job = new WorkspaceJob( "Importing Liferay SDK...")
+        {
+
+            @Override
+            public IStatus runInWorkspace( IProgressMonitor monitor ) throws CoreException
+            {
+                IStatus retval = StatusBridge.create( Status.createOkStatus() );
+
+                final SDK sdk = SDKUtil.createSDKFromLocation( PathBridge.create( sdkLocation ) );
+
+                SDKUtil.openAsProject( sdk );
+
+                return retval;
+            }
+        };
+
+        job.schedule();
+
+        return retval;
+    }
+    
+    public static IStatus validateBuildProperties( final SDK sdk )
```

I have checked that method name validate in idk class, but that method return the error status instead of warn status. I have to make this special method return warn status while the sdk build.{user.name}.properties file in this wizard.